sysutils/salt/Portfile

```diff
@@ -29,31 +29,78 @@ if {$subport eq $name} {
-
     notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api."
-
-    variant python34 conflicts python35 python36 description {experimental python-3.4 support} {
+    
+    default_variants	+python27 +zmq
+    
+    variant python27 conflicts python34 python35 python36 description {experimental python-3.5 support} {
+		python.default_version 27
+		depends_build	    port:py${python.version}-setuptools
+	
+		depends_lib-append  port:py${python.version}-yaml \
+							port:py${python.version}-jinja2 \
+							port:py${python.version}-msgpack
+							
+		notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api."
+    }
+    
+    variant python34 conflicts python27 python35 python36 description {experimental python-3.4 support} {
 		python.default_version 34
-		notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api. Support for Python 3 versions is experimental."
+		depends_build	    port:py${python.version}-setuptools
+	
+		depends_lib-append  port:py${python.version}-yaml \
+							port:py${python.version}-jinja2 \
+							port:py${python.version}-msgpack
+							
+		notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api. " \
+				"Support for Python 3 versions is experimental."
     }
-    variant python35 conflicts python34 python36 description {experimental python-3.5 support} {
+    
+    variant python35 conflicts python27 python34 python36 description {experimental python-3.5 support} {
 		python.default_version 35
-		notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api. Support for Python 3 versions is experimental."
+		depends_build	    port:py${python.version}-setuptools
+	
+		depends_lib-append  port:py${python.version}-yaml \
+							port:py${python.version}-jinja2 \
+							port:py${python.version}-msgpack
+							
+		notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api. " \
+				"Support for Python 3 versions is experimental."
     }
-    variant python36 conflicts python34 python35 description {experimental python-3.6 support} {
+    
+    variant python36 conflicts python27 python34 python35 description {experimental python-3.6 support} {
 		python.default_version 36
-		notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api. Support for Python 3 versions is experimental."
+		depends_build	    port:py${python.version}-setuptools
+	
+		depends_lib-append  port:py${python.version}-yaml \
+							port:py${python.version}-jinja2 \
+							port:py${python.version}-msgpack
+							
+		notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api. " \
+				"Support for Python 3 versions is experimental."
+    }
+    
+    variant zmq description {0mq TCPls  transport support} {
```

Comment:
Want to enable (encourage) testing of new Python3 support without forcing it. The depends are all python version specific. I am also a "convention over configuration" proponent, so I understand. In this case, until Python3 support is golden from upstream support, I think variants are a good thing. Since variants were needed, I thought it would be low-impact to add a few more, as optional (but delcared) dependencies. If those are bad I can take those out, but I don't see other packages depending on salt with those variants.

I'm aware of the testing complexity. I added a test phase so I could validated the depends were working.